Specs/Core/Matrix3Spec.js

```diff
@@ -107,29 +109,60 @@ defineSuite([
         var cPiOver2 = Math.cos(CesiumMath.PI_OVER_TWO);
 
         var tmp = Cartesian3.multiplyByScalar(new Cartesian3(0.0, 0.0, 1.0), sPiOver4, new Cartesian3());
-        var quatnerion = new Quaternion(tmp.x, tmp.y, tmp.z, cPiOver4);
+        var quaternion = new Quaternion(tmp.x, tmp.y, tmp.z, cPiOver4);
         var expected = new Matrix3(cPiOver2, -sPiOver2, 0.0, sPiOver2, cPiOver2, 0.0, 0.0, 0.0, 1.0);
         var result = new Matrix3();
-        var returnedResult = Matrix3.fromQuaternion(quatnerion, result);
+        var returnedResult = Matrix3.fromQuaternion(quaternion, result);
+        expect(result).toBe(returnedResult);
+        expect(returnedResult).toEqualEpsilon(expected, CesiumMath.EPSILON15);
+    });
+
+    it('fromHeadingPitchRoll works without a result parameter', function() {
+        var sPiOver4 = Math.sin(CesiumMath.PI_OVER_FOUR);
+        var cPiOver4 = Math.cos(CesiumMath.PI_OVER_FOUR);
+        var sPiOver2 = Math.sin(CesiumMath.PI_OVER_TWO);
+        var cPiOver2 = Math.cos(CesiumMath.PI_OVER_TWO);
+
+        var tmp = Cartesian3.multiplyByScalar(new Cartesian3(0.0, 0.0, 1.0), sPiOver4, new Cartesian3());
+        var quaternion = new Quaternion(tmp.x, tmp.y, tmp.z, cPiOver4);
+        var headingPitchRoll = HeadingPitchRoll.fromQuaternion(quaternion);
+        var expected = new Matrix3(cPiOver2, -sPiOver2, 0.0, sPiOver2, cPiOver2, 0.0, 0.0, 0.0, 1.0);
+
+        var returnedResult = Matrix3.fromHeadingPitchRoll(headingPitchRoll);
+        expect(returnedResult).toEqualEpsilon(expected, CesiumMath.EPSILON15);
+    });
+
+    it('fromHeadingPitchRoll works with a result parameter', function() {
+        var sPiOver4 = Math.sin(CesiumMath.PI_OVER_FOUR);
+        var cPiOver4 = Math.cos(CesiumMath.PI_OVER_FOUR);
+        var sPiOver2 = Math.sin(CesiumMath.PI_OVER_TWO);
+        var cPiOver2 = Math.cos(CesiumMath.PI_OVER_TWO);
+
+        var tmp = Cartesian3.multiplyByScalar(new Cartesian3(0.0, 0.0, 1.0), sPiOver4, new Cartesian3());
+        var quaternion = new Quaternion(tmp.x, tmp.y, tmp.z, cPiOver4);
+        var headingPitchRoll = HeadingPitchRoll.fromQuaternion(quaternion);
+        var expected = new Matrix3(cPiOver2, -sPiOver2, 0.0, sPiOver2, cPiOver2, 0.0, 0.0, 0.0, 1.0);
+        var result = new Matrix3();
+        var returnedResult = Matrix3.fromHeadingPitchRoll(headingPitchRoll, result);
         expect(result).toBe(returnedResult);
         expect(returnedResult).toEqualEpsilon(expected, CesiumMath.EPSILON15);
     });
 
     it('fromScale works without a result parameter', function() {
         var expected = new Matrix3(
-                7.0, 0.0, 0.0,
-                0.0, 8.0, 0.0,
-                0.0, 0.0, 9.0);
+            7.0, 0.0, 0.0,
```

Comment:
No need for all these `Matrix3` construction whitespace changes; it makes the diff harder to follow.